lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -288,24 +288,26 @@ fPulseWattage := SEL(fResponsivity <> 0, 0, (fVoltage - fBackgroundVoltage) / fR
 fCalibMJ := SEL(fFrequency <> 0, -9999, fPulseWattage / fFrequency);
 
 // FF in case voltage is high enough to damage power meter based on wavelength
-{IF defined (K)}
-    CASE REAL_TO_UINT(PMPS_GVL.stCurrentBeamParameters.neV) OF
-        0 .. uSoftWavelengthEdge1L: bOverAllowableEnergy := fCalibMJ > 2;
-        uSoftWavelengthEdge1 .. uSoftWavelengthEdge2L : bOverAllowableEnergy := fCalibMJ > 2;
-        uSoftWavelengthEdge2 .. uSoftWavelengthEdge3L: bOverAllowableEnergy := fCalibMJ > 4;
-        uSoftWavelengthEdge3 .. uSoftWavelengthEdge4L: bOverAllowableEnergy := fCalibMJ > 0.5;
-        uSoftWavelengthEdge4 .. 65535: bOverAllowableEnergy := fCalibMJ > 1;
-    END_CASE
-{ELSIF defined (L)}
-    CASE REAL_TO_UINT(PMPS_GVL.stCurrentBeamParameters.neV) OF
-        uHardWavelengthEdge1 .. uHardWavelengthEdge2L: bOverAllowableEnergy := fCalibMJ > 10;
-        uHardWavelengthEdge2 .. uHardWavelengthEdge3L: bOverAllowableEnergy := fCalibMJ > 4;
-        uHardWavelengthEdge3 .. 65535: bOverAllowableEnergy := fCalibMJ > 10;
-    END_CASE
-{END_IF}
-FFO(i_xOK := NOT bOverAllowableEnergy,
-    i_DevName := sDeviceName,
-    io_fbFFHWO := fbFFHWO);
+IF rTrig_Background.Q THEN
+    {IF defined (K)}
+        CASE REAL_TO_UINT(PMPS_GVL.stCurrentBeamParameters.neV) OF
+            0 .. uSoftWavelengthEdge1L: bOverAllowableEnergy := fCalibMJ > 2;
+            uSoftWavelengthEdge1 .. uSoftWavelengthEdge2L : bOverAllowableEnergy := fCalibMJ > 2;
+            uSoftWavelengthEdge2 .. uSoftWavelengthEdge3L: bOverAllowableEnergy := fCalibMJ > 4;
+            uSoftWavelengthEdge3 .. uSoftWavelengthEdge4L: bOverAllowableEnergy := fCalibMJ > 0.5;
+            uSoftWavelengthEdge4 .. 65535: bOverAllowableEnergy := fCalibMJ > 1;
+        END_CASE
+    {ELSIF defined (L)}
+        CASE REAL_TO_UINT(PMPS_GVL.stCurrentBeamParameters.neV) OF
+            uHardWavelengthEdge1 .. uHardWavelengthEdge2L: bOverAllowableEnergy := fCalibMJ > 10;
+            uHardWavelengthEdge2 .. uHardWavelengthEdge3L: bOverAllowableEnergy := fCalibMJ > 4;
+            uHardWavelengthEdge3 .. 65535: bOverAllowableEnergy := fCalibMJ > 10;
+        END_CASE
+    {END_IF}
+    FFO(i_xOK := NOT bOverAllowableEnergy,
+        i_DevName := sDeviceName,
+        io_fbFFHWO := fbFFHWO);
```

Comment:
An `FB_FastFault` registers itself on the first call, so by encasing it inside the IF statement here the registration is deferred.
It's probably better for FFs to be outside of the IF statement so that they will be registered on the first cycle and visible in pmps-ui right away (as ok/not faulting).

I think the cleaner way to do this is to keep much of the original formulation and then set `i_xOK` to something like `bOut OR NOT bOverAllowableEnergy`, defining `bOut` appropriately